### ISSUE TITLE
Submission Feedback

### ISSUE THREAD

**Issue by davidhorak** [powerdream5/django#1]
Hi @powerdream5,
Thank you for your submission, overall a great submission 🚀 , below are some notes.

Tests
- Could we add some simple tests? Just to showcase an approach.

#### Areas of improvements

> These are notes upon which you can optionally update the submission. Also, this is not a comprehensive list, rather some high level notes, if you would feel that you can improve the solution by other changes, making the implementation more suitable to be exposed to real users, please do so.

Input Validation
- I was able to enter "sample@domain" as email, only after submission to server, it was rejected with "Enter a valid email address."
- I was able to send the form (skipping the UI) directly with "bogus" phone number.
- Clicking on submit button will display only a single validation error, therefore I'm forced to address each validation error at time, and the hit submit again.

UI
- The role options, when text hovered, it does not have a pointer cursor (but actionable), and the radio button is not highlighted.

UX
- How we would handle a long list of members on the listing page?
- How we would better communicate to user that an action has been successful, .e.g: deleting a member. 

Accessibility
- We are missing H1
- Some of the actionable elements are missing titles, hence I am forced to click and trial understand what is action taken.

Data Integrity
- Accessing non expected URL, e.g.: http://localhost:8000/bogus, gives me a hard 404 server error.
- I was able to delete all users, how we would prevent system to be in the dead-end state?

Code
- Templates are mixed with inline JS, it would great to isolate them.

Kind regards,
David (Instawork)

**Comment by powerdream5** [comment]
Hi @davidhorak,

I'm truly impressed by the detailed review you provided. It has inspired me greatly. I plan to implement your suggestions over the weekend and will keep you updated.

Regarding the tests, I did indeed add the unit tests. I just included [instructions for running the tests](https://github.com/powerdream5/django?tab=readme-ov-file#unit-testing) in the documentation.

Best,
Wu

**Comment by davidhorak** [comment]
> Regarding the tests, I did indeed add the unit tests. I just included [instructions for running the tests](https://github.com/powerdream5/django?tab=readme-ov-file#unit-testing) in the documentation.

My bad! Thanks for resurfacing @powerdream5 

Kind regards,
David

**Comment by powerdream5** [comment]
Hi @davidhorak, 

I've implemented the changes based on your feedback:

1. Input Validation: I've improved the input validation by incorporating JavaScript validation, making it much more user-friendly than the default web browser validation.

2. UI: The role selection radio buttons now highlight upon hover.

3. UX: Added pagination to the list and implemented a success confirmation message for adding, updating, or deleting a member.

4. Accessibility: I've reviewed the pages using the WAVE tool (https://chromewebstore.google.com/detail/wave-evaluation-tool/jbbplnpkjmmeebjpijfedlgcdilocofh).

5. Data Integrity: Introduced a 404 error page. Regarding the feature to delete all users, normally, permission checks and safeguards against deleting one's own account would be necessary. However, for this demonstration project, I haven't implemented any permission checks.

6. Code: I've separated the inline JavaScript code into its own file.

Thank you once again for your feedback.  Welcome any additional comments.

Best,
Wu Cao
